doc/README.md

```diff
@@ -1,4 +1,60 @@
 KOS_DOC
 =======
+The documents are generated using Sphinx restructured text, with the ReadTheDocs
+theme.
 
-New documentation source for Sphinx/reStructuredText format.
+#Getting started on Windows
+
+(For this example, the KOS repository is assumed to be located at `C:\KOS`,
+you should adjust your path based on your actual repository location)
+
+1. If you don't already have Python installed, install the latest version in the
+  2.7 series.  At the time of this writing, 2.7.11 was the most current version.
+2. You can verify the installation and version of Python (and pip) by issuing
+  the following commands from the command line:
+  ```
+  C:\>python --version
+  Python 2.7.11
+  C:\>pip --version
+  pip
+  ```
+3. Install the Sphinx engine
+  ```
+  C:\>pip install sphinx
+  ```
+4. Ensure that the read the docs template is installed.
+  ```
+  C:\>pip install sphinx_rtd_theme
+  ```
+5. Switch to the docs directory and run the make batch file:
+  ```
+  C:\>cd KOS\docs
+  C:\KOS\docs>make html
+  ```
+6. Review the output for errors and warnings.  In the above example you would
+  find the compiled html files at `C:\KOS\docs\gh-pages`
+7. (Optional) You may browse the generated html using file urls, or by using
+  Python's included SimpleHTTPServer:
+  ```
+  C:\KOS\docs>cd gh-pages
+  C:\KOS\docs\gh-pages>python -m SimpleHTTPServer 8000
+  Serving HTTP on 0.0.0.0 port 8000...
+  ```
+  At which point you can point your browser to `http://localhost:8000`
+
+#Getting started on Linux
+1. TODO: WRITE LINUX INSTRUCTIONS WITH APT-GET AND PIP
+
+#Publishing
+1. Clone KSP-KOS/KOS gh-pages branch if not done already, pull the most recent
+  version from the upstream repository if you already have a clone.
+2. The clone should either be in a different folder from the primary
+  repository or you need to copy the rendered `doc/gh-pages` contents to a
+  temporary folder
+3. Delete the contents of the repository root, except the `.git` folder,
+  `.gitatributes` file, and `.nojekyll` file.
```

Comment:
Using stash won't work because of how file tracking is managed, so that idea is out.

>"Delete everything, recursing through all directories,except for those files and directories that begin with a dot, for example the .git folder, .gitattributes, and so on."

>I was saying you should because it catches everything, and it prevents accidentally adding files you don't want to add if you're on a unix system, where often tools add dot files all the time willy-nilly...

The way I read those two statements is contradictory.  First you tell me to give instructions to delete everything without a leading `.`, because you want to preserve those files.  Then you say that unix adds these `.` files every where and you want to avoid committing these extra files.

My understanding is that we want to only keep the files `.gitattributes` and `.nojekyll` and the directory `.git`.  Nothing else. Then we copy the contents of the gh-pages folder into the root.

If they didn't clean the working directory properly, we don't want to keep the `.ci` folder, nor do we want to include a random `.` file that their system added to the directory, which isn't tracked and therefore not deleted by the checkout command.  (this is probably the biggest issue, untracked files aren't touched by git commands).  So I think we do want to use a white list, so that there isn't a random file or folder kept in the folder which shouldn't be in the repo.  If we need to preserve additional files in the future, we would just need to update the list.

If what you really want me to do is tell them explicitly to delete all files with a leading `.`, excluding those on the list, then I will need you to give me an appropriate command to do so.